[Enhancement] Refactor high-level APIs (#1410)
open-mmlab/mmediting: demo/README.md

# MMEditing Demo

> There are some mmediting demos in this folder. We provide python command line usage here to run these demos and more guidance could also be found on the [web page](https://mmediting.readthedocs.io/en/dev-1.x/user_guides/3_inference.html)

Reviewer: ```suggestion
There are some mmediting demos in this folder. We provide python command line usage here to run these demos and more guidance could also be found on the [documentation](https://mmediting.readthedocs.io/en/dev-1.x/user_guides/3_inference.html)
```

Author: thanks. resolved :smiley:

## Download sample images or videos

We prepared some images and videos for you to run demo with. After MMEdit is well installed, you could use demos in this folder to infer these data. Download at here (url) and extract it to MMEdit root path.

```shell
cd mmediting
wget url
unzip resources.zip
```

## MMEditing inference demo

You can use the following commands to perform inference with a MMEdit model.

> Usage of python API can be found in this [tutotial](demo/mmediting_inference_tutorial.ipynb).

Reviewer: ```suggestion
Usage of python API can be found in this [tutotial](./mmediting_inference_tutorial.ipynb).
```
```suggestion
Usage of python API can be found in this [tutotial](demo/mmediting_inference_tutorial.ipynb).
```

> Usage of python API can be found in this [tutotial](demo/mmediting_inference_tutorial.ipynb).

Reviewer: ```suggestion
Usage of python API can be found in this [tutotial](./mmediting_inference_tutorial.ipynb).
```

```

8. Video restoration

```shell
python demo/mmediting_inference_demo.py \
        --model-name edvr \
        --extra-parameters window_size=5 \
        --video resources/input/video_interpolation/QUuC4vJs_000084_000094_400x320.mp4 \
        --result-out-dir resources/output/video_restoration/demo_video_restoration_edvr_res.mp4
```

## Face restoration demo

You can use the following commands to test an face image for restoration.

```shell
python demo/restoration_face_demo.py \
    ${CONFIG_FILE} \
    ${CHECKPOINT_FILE} \
    ${IMAGE_FILE} \
    ${SAVE_FILE} \
    [--upscale-factor] \
    [--face-size] \
    [--imshow] \
    [--device ${GPU_ID}]
```

Examples:

```shell
python demo/restoration_face_demo.py \
    configs/glean/glean_in128out1024_4x2_300k_ffhq_celebahq.py \
    https://download.openmmlab.com/mmediting/restorers/glean/glean_in128out1024_4x2_300k_ffhq_celebahq_20210812-acbcb04f.pth \
    tests/data/image/face/000001.png \
    tests/data/pred/000001.png \
    --upscale-factor 4
```

## Other demos

These demos are duplicated with mmedting_inference_demo.py and may be removed in the future.

> - colorization_demo.py

Reviewer: ```suggestion
- [colorization_demo.py](./colorization_demo.py)
```

Author: thanks. resolved :smiley:

> - conditional_demo.py

Reviewer: ```suggestion
- [conditional_demo.py](./conditional_demo.py)
```

Author: thanks. resolved :smiley:

# MMEditing Demo

There are some mmediting demos in this folder. We provide python command line usage here to run these demos and more guidance could also be found in the [documentation](https://mmediting.readthedocs.io/en/dev-1.x/user_guides/3_inference.html)

Table of contents:

[1. Download sample images or videos](#1-download-sample-images-or-videos)

[2. MMEditing inference demo](#2-mmediting-inference-demo)

&#8195;    [2.1. Check supported tasks and models](#21-check-supported-tasks-and-models)

&#8195;    [2.2. Perform inference with command line](#22-perform-inference-with-command-line)

&#8195;      [2.2.1. Conditional GANs example](#221-conditional-gans)

&#8195;      [2.2.2. Inpainting example](#222-inpainting)

&#8195;      [2.2.3. Matting example](#223-matting)

&#8195;      [2.2.4. Image Super-Resolution example](#224-image-super-resolution)

&#8195;      [2.2.5. Image Translation example](#225-image-translation)

&#8195;      [2.2.6. Unconditional GANs example](#226-unconditional-gans)

&#8195;      [2.2.7. Video Interpolation example](#227-video-interpolation)

&#8195;      [2.2.8. Video Super-Resolution example](#228-video-super-resolution)

> [3. Face restoration demo](#3-face-restoration-demo):

Reviewer: ```suggestion
[3. Face restoration demo](#3-face-restoration-demo)
```

Author: sorry for this typo, resolved

[4. Other demos](#4-other-demos)

## 1. Download sample images or videos

We prepared some images and videos for you to run demo with. After MMEdit is well installed, you could use demos in this folder to infer these data.
Download with python script [download_inference_resources.py](./download_inference_resources.py).

```shell

> # see all resources

Reviewer: cd mmediting/demo

Author: resolved.

python download_inference_resources.py --print-all
# see all task types
python download_inference_resources.py --print-task-type
# see resources of one specific task
python download_inference_resources.py --print-task 'Inpainting'
# download all resources to default dir '../resources'
python download_inference_resources.py
# download resources of one task
python download_inference_resources.py --task 'Inpainting'
# download to the directory you want
python download_inference_resources.py --root-dir '../your_dir'
```

## 2. MMEditing inference demo

### 2.1 Check supported tasks and models

> print all supported models for inference.

Reviewer: cd mmediting

Author: resolved.
